### PR TITLE
Use golang.org/x/sys instead of self-implemented detector

### PR DESCRIPTION
**Pull request by im-kulikov**
- Ignore vendor and tzsum binary
- Update golang and alpine images
- Add dependency target for Makefile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.11-alpine3.8 as builder
+FROM golang:1-alpine as builder
```
Comment (realloc) [inline Dockerfile]: Why did you omit the exact version here?

Comment (im-kulikov) [reply Dockerfile]: just to use the latest version of the golang

File: tz/hash.go [hunk 2]
```diff
@@ -67,9 +60,9 @@ func NewWith(impl Implementation) hash.Hash {
 
 // New returns a new hash.Hash computing the Tillich-Zémor checksum.
 func New() hash.Hash {
-	if hasAVX2 {
+	if cpu.X86.HasAVX2 {
```
Comment (fyrchik) [inline tz/hash.go]: cpu.X86.HasAVX2 && cpu.X86.HasAVX

Comment (im-kulikov) [reply tz/hash.go]: see [`x/sys` package](https://github.com/golang/sys/blob/86b910548bc16777f40503131aa424ae0a092199/cpu/cpu_x86.go#L50)

Comment (fyrchik) [reply tz/hash.go]: I don't think that's enough, because our AVX2 implementation uses AVX instructions, not only X*/Y* registers.

**Comment by fyrchik** [comment]
Could you also split this into 3 commits: 1 for `.gitignore`, 1 for `Makefile` and another for the actual module changes? It will be easier to read.
Otherwise, lgtm.

**Comment by im-kulikov** [comment]
> Could you also split this into 3 commits: 1 for `.gitignore`, 1 for `Makefile` and another for the actual module changes? It will be easier to read.
> Otherwise, lgtm.

I suggest making the next commits:
- .gitignore as you proposed
- Makefile + Dockerfile + benchmark as separated commit
- changes for the module as you proposed

What you think, @fyrchik ?

**Pull request merged**